Gems/Atom/Utils/Code/Source/ImageComparison.cpp

```diff
@@ -12,6 +12,16 @@ namespace AZ
 {
     namespace Utils
     {
+        int16_t CalcMaxChannelDifference(AZStd::array_view<uint8_t> bufferA, AZStd::array_view<uint8_t> bufferB, size_t index)
+        {
+            // We use the max error from a single channel instead of accumulating the error from each channel.
+            // This normalizes differences so that for example black vs red has the same weight as black vs yellow.
+            const int16_t diffR = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index]) - aznumeric_cast<int16_t>(bufferB[index])));
+            const int16_t diffG = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 1]) - aznumeric_cast<int16_t>(bufferB[index + 1])));
+            const int16_t diffB = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 2]) - aznumeric_cast<int16_t>(bufferB[index + 2])));
```

Comment:
Does abs change the result type? is the outer static cast needed?

Reply:
Yes it's needed, abs returns an int.